src/dune

```diff
@@ -0,0 +1,43 @@
+;; generate menhir tokens
+(rule
+ (targets Tokens.ml)
+ (deps Menhir_parser.mly)
+ (action (run menhir --only-tokens %{deps} --base Tokens)))
+
+;; generate lexer -- temporary hack, see https://gitlab.inria.fr/fpottier/menhir/issues/6
+(rule
+ (targets Lexer.ml)
+ (deps    (:f Lexer.cppo.ml) Tokens.ml)
+ (action  (run cppo %{f} -n -o %{targets})))
+
+(rule
+ (targets Lexer.mli)
+ (deps    (:f Lexer.cppo.mli) Tokens.ml)
+ (action  (run cppo %{f} -n -o %{targets})))
+
+;; generate parser
+(menhir
+ (modules Menhir_parser)
+ (flags (--external-tokens Lexer)))
+
+(library
+ (name        ocaml_parsing)
+ (public_name ocaml_parsing)
+ (wrapped true)
+ ;;(deps (Tokens))
+ (libraries 
+   base
+   stdio
+   sedlex
+   compiler-libs.common
+   menhirLib
+ )
+ (virtual_deps cppo menhir)
+ (preprocess (pps ppx_deriving.std ppx_enumerate sedlex.ppx))
+)
+
+;; Ignore unused rec flag warning
```

Comment:
Instead of ignoring the warning, can we fix the issue?

Reply:
I tried debugging it. It seems the sedlex PPX is generating code with an unused rec flag. It might be because of the simple lexer because I don't have this problem in my other project where I use this boilerplate.

Reply:
This is the relevant part of the code generated by the sedlex PPX. Notice I added `[@ocaml.warning "-39"]` to the `token:Lexer.ml` function.

```
$ cd _build/default
$ .ppx/ppx_deriving.std+ppx_enumerate+sedlex.ppx/ppx.exe src/Lexer.ml | less
let token buf =
  ((garbage buf;    (let rec __sedlex_state_0 =
       function
       | buf ->
           (match __sedlex_partition_7 (Sedlexing.next buf) with
            | 0 -> 0
            | 1 -> 1
            | 2 -> 2
            | _ -> Sedlexing.backtrack buf) in
     Sedlexing.start buf;
     (match __sedlex_state_0 buf with
      | 0 -> EOF
      | 1 -> LPAR
      | 2 -> RPAR
      | _ -> illegal buf (Char.chr (next buf)))))
  [@ocaml.warning "-39"])
```